Sources/ApolloCodegenLib/ApolloCodegenConfiguration.swift

```diff
@@ -6,19 +6,26 @@ public struct ApolloCodegenConfiguration {
   public struct FileInput {
     /// Local path to the GraphQL schema file. Can be in JSON or SDL format.
     public let schemaPath: String
-    /// Glob of files to search for GraphQL operations. This should be used to find queries and any client schema extensions.
-    public let glob: String
+    /// A comma-delimited string of path matching patterns to search for GraphQL operations, such as queries and mutations.
+    /// 
+    /// Each path matching pattern can include the following characters:
+    /// - `*` matches 0 or more characters in a single path portion, eg: `*.graphql`
+    /// - `?` matches 1 character, eg: `file-?.graphql`
+    /// - `**` matches zero or more directories and subdirectories searching for matches, eg: `**/*.graphql`
+    public let includes: String
```

Comment:
Is this worth setting up as an enum with associated values instead of a globstar string? Then you can have an array of them and have things defined more clearly

Reply:
Your comment is along the lines of what I'm still working on; namely dropping the term 'glob' (because it's difficult to express its full ability and that ability is different on each system) and turning it into comma-delimited list of path matching patterns which allow for some special characters.

I'm interested to hear more about your enum idea though..

Reply:
I was thinking something like:

```swift
enum SearchOption {
   case shallow(value: String)
   case matchingAnyCharacter(in: String)
   case anyMatchInSubdirectories(using: String)
   case excluding(using: String)
}
```

Although it'd need to be an option set to allow combinations, and I can't remember if option  sets work with enums with associated values right now. Will poke at this some on monday.

Reply:
With https://github.com/apollographql/apollo-ios/pull/2041/commits/3723bbc4b8760543e14ddbef37e53d696f81cc07 it's an array of strings. Which gets rid of some logic and is a _slightly_ more Swifty way to pass these values in.

_I'm not yet sold on the enum idea._

Reply:
Yeah on further investigation it looks like turning it to an option set won't work, at least not without a bunch of heavy lifting: 

<img width="471" alt="Screen Shot 2021-11-29 at 11 53 46 AM" src="https://user-images.githubusercontent.com/1976498/143918516-cafb2871-edd2-4c33-8f59-53b401e7207e.png">

And I don't know if it'd work well if you just passed in an array of `SearchOption`. 

In general I'm not a *giant* fan of things like globstar and regex because I think they make it a lot harder for people who aren't familiar with those structures to understand what's happening. But they do exist for a reason...

Reply:
I agree they do get complicated to explain when you start trying to express the full ability of them. That's why I've reduced it to a few special characters that should give users all they need to tell us where to find the operation files.

```
  /// - `*` matches everything but the directory separator (shallow), eg: `*.graphql`
  /// - `?` matches any single character, eg: `file-?.graphql`
  /// - `**` matches all subdirectories (deep), eg: `**/*.graphql`
  /// - `!` excludes any match, eg: `*.graphql,!file.graphql`
```